src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs

```diff
@@ -79,49 +45,62 @@ internal string AccessToken
             }
         }
 
-        internal ServerCertificateValidationCallback ServerCertificateValidationCallback
+        public override int GetHashCode()
         {
-            get
-            {
-                return _serverCertificateValidationCallback;
-            }
+            return _hashValue;
         }
 
-        internal ClientCertificateRetrievalCallback ClientCertificateRetrievalCallback
+#if NETFRAMEWORK
+        private readonly ServerCertificateValidationCallback _serverCertificateValidationCallback;
+        private readonly ClientCertificateRetrievalCallback _clientCertificateRetrievalCallback;
+        private readonly SqlClientOriginalNetworkAddressInfo _originalNetworkAddressInfo;
+
+        private SqlConnectionPoolKey(SqlConnectionPoolKey key) : base(key)
         {
-            get
-            {
-                return _clientCertificateRetrievalCallback;
-            }
+            _credential = key.Credential;
+            _accessToken = key.AccessToken;
+            _serverCertificateValidationCallback = key._serverCertificateValidationCallback;
+            _clientCertificateRetrievalCallback = key._clientCertificateRetrievalCallback;
+            CalculateHashCode();
         }
```

Comment:
This function could be taken out to the common section. Plz, follow the [old commit](https://github.com/dotnet/SqlClient/pull/1292/commits/4d1f09530317d3ac5264b183ed1dd94db7af7041#diff-813dd48970098a3e3745790fdd8a33f0a9d98162936ebe899fbec804d91f19edR51) and avoid redundancy.